### PR TITLE
Lead provider School Reporting: CSV full journey

### PR DESCRIPTION
**Pull request by BroiSatse**
This PR is a final review and refactor of the whole School Reporting journey for Lead Provider user using CSV file.

Saying it is final does not mean it is in any way ideal, however it comprises a number of refactors turning the entire journey into a well-defined and encapsulated module. Files has been moved around, forms been merged together into one coherent form, controller logic has been altered to respect their autonomy. It is possibly best to review this by checking it out locally to see the final folder structure.

### CODE REVIEW

File: app/views/lead_providers/report_schools/base/start.html.erb [hunk 1]
```diff
@@ -5,18 +5,18 @@
...
           Schools
         <% end %>
       <% end %>
     <% end %>
 
     <h1 class="govuk-heading-xl">You can only confirm schools for the cohort that starts in the next academic year
-      (<%= Cohort.current.display_name %>).</h1>
+      (<%= report_schools_form.cohort.display_name %>).</h1>
 
     <p>If you want to add schools to your 2021 cohort after 31 October 2021, you need to contact:
```
Comment (ethanmills) [inline app/views/lead_providers/report_schools/base/start.html.erb:17]: If we're making the cohort variable, we need to fix this too

File: app/controllers/lead_providers/report_schools/csv_controller.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          else
+            redirect_to action: :errors
+          end
+        else
+          render :show
+        end
+      end
+
+      def errors
+        partnership_csv_upload = PartnershipCsvUpload.find(session[:partnership_csv_upload_id])
```
Comment (ethanmills) [inline app/controllers/lead_providers/report_schools/csv_controller.rb:37]: Is there a reason you can't also do 

```
report_schools_form.school_ids = partnership_csv_upload.valid_schools.pluck(:id)
session.delete(:partnership_csv_upload_id)
```

here and remove `load_partnership_csv_upload` from confirm?

File: app/models/partnership_csv_upload.rb [hunk 3]
```diff
@@ -47,6 +47,7 @@ def find_school_errors
       school = School.includes(:partnerships).find_by(urn: urn)
       school = School.find_by(urn: urn.sub!(/^[0]+/, "")) if school.blank?
 
+      # TODO: Should we dependent on `Cohort.current` here or should this model reference cohort it was used for?
```
Comment (ethanmills) [inline app/models/partnership_csv_upload.rb:50]: let's change it to be the cohort on the form

**Review by ethanmills — APPROVED** [review]
Looks good, couple of minor things

File: app/controllers/lead_providers/report_schools/csv_controller.rb (hunk 2 above)
Comment (BroiSatse) [reply app/controllers/lead_providers/report_schools/csv_controller.rb:37]: @ethanmills AH, that's exactly why I've added `proceed` action. The whole `load_partnership_csv_upload` is now obsolete. Not sure Why I haven't remove it! :D

File: app/views/lead_providers/report_schools/base/start.html.erb (hunk 1 above)
Comment (BroiSatse) [reply app/views/lead_providers/report_schools/base/start.html.erb:17]: Nice catch!

File: app/models/partnership_csv_upload.rb (hunk 3 above)
Comment (BroiSatse) [reply app/models/partnership_csv_upload.rb]: Done!

**Pull request merged**